packages/database/src/migrations/20220630013151-LinkSyncServiceToSurvey-modifies-schema.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  await db.runSql(
+    `DROP TRIGGER IF EXISTS data_service_sync_group_trigger ON data_service_sync_group`,
+  );
+
+  // Add new code column which should match current syncGroupCode
+  await db.runSql(`ALTER TABLE data_service_sync_group ADD COLUMN data_group_code text`);
+
+  await db.runSql(`
+    UPDATE data_service_sync_group 
+    SET 
+      data_group_code = trim(both '"' from config->>'internalSurveyCode'),
+      code = trim(both '"' from config->>'internalSurveyCode'),
+      config = (config || jsonb_build_object('syncGroupCode', code)) - 'internalSurveyCode'
+  `);
```

Comment:
@IgorNadj reworked this as discussed.

Note: the kobo sync logic won't actually work right now (due to a mismatch between the code in the `sync_group` table's `config` column and the current values) however that will be fixed up in the [pt2 PR](https://github.com/beyondessential/tupaia/pull/4089)